File: backend/src/Workora.Application/Features/Loans/DTOs/LoanDto.cs
using Workora.Domain.Enums;
using Workora.Shared.Responses;

namespace Workora.Application.Features.Loans.DTOs;

/// <summary>
/// Data transfer object representing an employee loan account.
/// </summary>
public record LoanDto(
    int Id,
    Guid Uuid,
    int EmployeeId,
    string? EmployeeName,
    string? EmployeeCode,
    LoanType LoanType,
    decimal PrincipalAmount,
    int TenureMonths,
    decimal MonthlyEmi,
    decimal TotalRepaid,
    decimal RemainingBalance,
    DateOnly DisbursementDate,
    LoanStatus Status,
    string Reason,
    int? ApprovedByUserId,
    DateTimeOffset? ApprovedAt,
    string? RejectionReason,
    DateTimeOffset CreatedAt);

File: frontend/workora-ui/src/app/data/dtos/loan.dto.ts
export interface LoanEmiScheduleDto {
  id: number;
  uuid: string;
  loanRecordId: number;
  installmentNumber: number;
  dueDate: string;
  emiAmount: number;
  principalComponent: number;
  interestComponent: number;
  isPaid: boolean;
  paidAt?: string | null;
  payrollRunId?: number | null;
}

export interface LoanDto {
  id: number;
  uuid: string;
  employeeId: number;
  employeeName?: string | null;
  employeeCode?: string | null;
  loanType: string;
  principalAmount: number;
  tenureMonths: number;
  monthlyEmi: number;
  totalRepaid: number;
  remainingBalance: number;
  disbursementDate: string;
  status: string;
  reason: string;
  approvedByUserId?: number | null;
  approvedAt?: string | null;
  rejectionReason?: string | null;
  createdAt: string;
}

export interface ApplyLoanRequestDto {
  employeeId: number;
  loanType: string;
  principalAmount: number;
  tenureMonths: number;
  reason: string;
}
